elytron/ELY-1686-legacy-properties-files-into-FileSystemRealm-utility.adoc

```diff
@@ -0,0 +1,199 @@
+= ELY-1686 Add utility to port legacy properties files into a FileSystemRealm
+:author:            Justin Michael Cook
+:email:             jucook@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+Currently, converting legacy security properties files into an Elytron FileSystemRealm
+must be done manually, there should be a migration tool that automates this conversion.
+This task is to add such a tool to WildFly Elytron Tool.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1686[ELY-1686]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1093[EAP7-1093]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:msvehla@redhat.com[Martin Švehla]
+
+=== Affected Projects or Components
+
+* WildFly Elytron Tool
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* It should be possible to have legacy properties file converted into an Elytron
+FileSystemRealm automatically. This should be done by adding a command named
+FileSystemRealmCommand to the Elytron-Tool. This command will create a users directory from
+a given path and then use that directory to create a FileSystemRealm. The command will also
+configure the security-domain, http-authentication-factory, and application-security-domain.
+The command will provide a script file containing any additional WildFly CLI commands needed
+to fully register the FileSystemRealm in Elytron.
+
+* This command will require a users properties file, roles properties file, output directory,
+and password to be specified. The paths to these files and directory should be able to be
+relative to the administrator's current directory or should be able to be absolute.
+Optionally, it will require the name of the filesystem-realm, security-domain,
+http-authentication-factory, and application-security domain to be specified. If these
+options are not specified, then they will be generated using the inputted file names.
+It should be possible to specify the password using any supported mechanism.
+
+** Command with only required options
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm --user-file/-u <userFile> --roles-file/-r <rolesFile>
+--output-location/-o <outDirectory> --password/-p <password>
+
+** Command with all possible options
+[source,bash]
+./bin/elytron-tool.sh filesystem-realm --user-file/-u <userFile> --roles-file/-r <rolesFile>
+--output-location/-o <outDirectory> --password/-p <password> [--filesystem-realm/-f] <name>
```

Comment:
To make it a bit more clear that the ```filesystem-realm``` parameter requires a name, it may be better to call it ```filesystem-realm-name```. Same with other parameters that take a name.